src/commands/deployWorkspaceProject/getDefaultValues/getDefaultContainerAppsResources/getDefaultResourcesFrom.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ContainerAppsAPIClient } from "@azure/arm-appcontainers";
+import type { ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { ext } from "../../../../extensionVariables";
+import { ContainerAppItem, ContainerAppModel } from "../../../../tree/ContainerAppItem";
+import { ManagedEnvironmentItem } from "../../../../tree/ManagedEnvironmentItem";
+import { createContainerAppsAPIClient } from "../../../../utils/azureClients";
+import { localize } from "../../../../utils/localize";
+import type { DeployWorkspaceProjectSettings } from "../../deployWorkspaceProjectSettings";
+import type { DefaultContainerAppsResources } from "./getDefaultContainerAppsResources";
+import { getResourcesFromContainerAppHelper, getResourcesFromManagedEnvironmentHelper } from "./getResourceHelpers";
+
+const noMatchingResources = {
+    resourceGroup: undefined,
+    managedEnvironment: undefined,
+    containerApp: undefined
+};
+
+export async function getContainerAppResourcesFromSettings(context: ISubscriptionActionContext, settings: DeployWorkspaceProjectSettings | undefined): Promise<DefaultContainerAppsResources> {
+    if (!settings || !settings.containerAppResourceGroupName || !settings.containerAppName) {
```

Comment:
This used to be existing code in `getDefaultContainerAppsResources`, I just moved it to a new file.